api/src/events.rs

```diff
@@ -165,6 +168,55 @@ pub async fn process_drop_minted_event(
     Ok(())
 }
 
+/// Processes a `MintTransfered` event and updates the corresponding entities in the database.
+///
+/// # Arguments
+///
+/// * `db` - A database connection object used to interact with the database.
+/// * `key` - A `TreasuryEventKey` representing the key of the event.
+/// * `payload` - A `MintTransfered` struct representing the payload of the event.
+/// # Errors
+///
+/// This function returns an error if it fails to update the mint owner or
+/// if it fails to inserts the nft transfer
+
+pub async fn process_mint_transfered_event(
+    db: Connection,
+    key: TreasuryEventKey,
+    payload: MintTransfered,
+) -> Result<()> {
+    let MintTransfered {
+        sender,
+        recipient,
+        tx_signature,
+        ..
+    } = payload;
+
+    let id = Uuid::from_str(&key.id)?;
+
+    let collection_mint = collection_mints::Entity::find_by_id(id)
+        .one(db.get())
+        .await
+        .context("failed to load collection mint from db")?
+        .context("collection mint not found in db")?;
+
+    let mut collection_mint_am: collection_mints::ActiveModel = collection_mint.into();
+    collection_mint_am.owner = Set(recipient.clone());
+    collection_mint_am.update(db.get()).await?;
+
+    let nft_transfer_am = nft_transfers::ActiveModel {
+        tx_signature: Set(tx_signature),
+        mint_id: Set(id),
+        sender: Set(sender),
+        receiver: Set(recipient),
+        ..Default::default()
+    };
+
+    nft_transfer_am.insert(db.get()).await?;
+
+    Ok(())
+}
```

Comment:
To be able to show a transfer in progress lets write the nft_transfer record in the mutation with out the tx_signature then update the record here with it.